src/varity/vcf_lift.clj

```diff
@@ -0,0 +1,62 @@
+(ns varity.vcf-lift
+  (:require [cljam.io.vcf.util.normalize :as norm]
+            [cljam.io.vcf.util :as vcf-util]
+            [cljam.io.sequence :as io-seq]
+            [cljam.util.sequence :as util-seq]
+            [cljam.util.chromosome :as chr]
+            [varity.chain :as ch]))
+
+(defn- calc-new-interval [chr beg end indexed-chains]
+  (some #(when-let [overlap-blocks
+                    (seq (ch/search-overlap-blocks beg end (:data %)))]
+           (let [start-offset
+                 (max 0 (- beg (:t-start (first overlap-blocks))))
+                 end-offset (max 0 (- (:t-end (last overlap-blocks))
+                                      end))
+                 to-start (+ (:q-start (first overlap-blocks)) start-offset)
+                 to-end (- (:q-end (last overlap-blocks)) end-offset)
+                 strand (get-in % [:header :q-strand])]
+             (if (= strand :reverse)
+               {:chr (get-in % [:header :q-name])
+                :start (inc (- (get-in % [:header :q-size]) to-end))
+                :end (inc (- (get-in % [:header :q-size]) to-start))
+                :strand strand}
+               {:chr (get-in % [:header :q-name])
+                :start to-start :end to-end :strand strand})))
+        (ch/search-containing-chains chr beg end indexed-chains)))
```

Comment:
I guess we need to compare scores if the query hits multiple chains.
Would you add tests for such cases?